x-pack/plugins/endpoint/server/routes/alerts.ts

```diff
@@ -5,27 +5,101 @@
  */
 
 import { IRouter } from 'kibana/server';
+import { RequestHandler } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema, TypeOf } from '@kbn/config-schema';
 
-import json from './sampledata.json';
+import {
+  getPagingProperties,
+  kibanaRequestToAlertListQuery,
+} from '../services/endpoint/alert_query_builders';
+
+import { AlertData, AlertResultList } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const ALERTS_ROUTE = '/api/endpoint/alerts';
+
+export const reqSchema = schema.object({
+  page_size: schema.number({ defaultValue: 10, min: 1, max: 10000 }),
+  page_index: schema.number({ defaultValue: 0, min: 0 }),
+});
+
+export function registerAlertRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const alertsHandler: RequestHandler<unknown, TypeOf<typeof reqSchema>> = async (
+    ctx,
+    req,
+    res
+  ) => {
+    try {
+      const queryParams = await getPagingProperties(req, endpointAppContext);
+      const reqBody = await kibanaRequestToAlertListQuery(queryParams, endpointAppContext);
+      const response = (await ctx.core.elasticsearch.dataClient.callAsCurrentUser(
+        'search',
+        reqBody
+      )) as SearchResponse<AlertData>;
+      return res.ok({ body: mapToAlertResultList(endpointAppContext, queryParams, response) });
+    } catch (err) {
+      return res.internalError({ body: err });
+    }
+  };
 
-export function registerAlertRoutes(router: IRouter) {
   router.get(
     {
-      path: '/api/endpoint/alerts',
-      validate: false,
+      path: ALERTS_ROUTE,
+      validate: {
+        query: reqSchema,
+      },
       options: { authRequired: true },
     },
-    async (context, req, res) => {
-      try {
-        return res.ok({
-          body: json,
-          headers: {
-            'Content-Type': 'application/json',
-          },
-        });
-      } catch (err) {
-        return res.internalError({ body: err });
-      }
-    }
+    alertsHandler
   );
+
+  router.post(
+    {
+      path: ALERTS_ROUTE,
+      validate: {
+        body: reqSchema,
+      },
+      options: { authRequired: true },
+    },
+    alertsHandler
+  );
+}
+
+function mapToAlertResultList(
+  endpointAppContext: EndpointAppContext,
+  queryParams: Record<string, any>,
+  searchResponse: SearchResponse<AlertData>
+): AlertResultList {
+  interface Total {
```

Comment:
Could we move this type to `server/types.ts`? Also, this type could probably be shared with the endpoint list handlers.

Reply:
@nnamdifrankie is working on a refactor PR for things like this, but this PR might not be merged in time. We'll have to do this later, I think...

Reply:
We plan to refactor things that are common so that we have the same field spelling. But I will be wary to share too much things, that can deviate base on different streams development. Single responsibility.